### ISSUE TITLE
Board editor: Top and bottom planes can't be fully filled. In most cases bottom plane can't be filled at all for some reasons.

### ISSUE THREAD

**Issue by azhizhinov** [LibrePCB/LibrePCB#974]
##### VERSION / OS / ENVIRONMENT

```
LibrePCB Version: 0.1.6
Git Revision: c4f316428
Qt Version: 5.15.3 (built against 5.15.3)
Operating System: Gentoo/Linux
...
```

##### SUMMARY

Top and bottom planes can't be fully filled. In most cases bottom plane can't be filled at all for some reasons.

##### STEPS TO REPRODUCE

Project .lppz has been attached.

##### EXPECTED RESULTS

Planes are filled.

##### ACTUAL RESULTS

Top plane has been filled partially. Bottom plane not filled at all.

##### ADDITIONAL INFORMATION

![20220501 LIBREPCB ORG 00](https://user-images.githubusercontent.com/13044908/166137724-73147773-a9b3-4e90-ab5e-3520a5c134b9.jpg)
![20220501 LIBREPCB ORG 01](https://user-images.githubusercontent.com/13044908/166137730-b37b68f5-6453-4cce-b4d4-ac794a1975e8.jpg)
[For_Science_Keyboard.lppz.zip](https://github.com/LibrePCB/LibrePCB/files/8598504/For_Science_Keyboard.lppz.zip)

**Comment by ubruhin** [comment]
Thanks for the detailed bug report. However, I cannot reproduce it :see_no_evil: 

Do you know whether your LibrePCB binary was built with the bundled polyclipping library or with polyclipping from the package manager (using `UNBUNDLE_POLYCLIPPING`)? If unbundled, what version was used?

Alternatively you could try to reproduce it with our official AppImage.

**Comment by ktrace** [comment]
-DUNBUNDLE_QUAZIP=ON
-DUNBUNDLE_MUPARSER=ON
Bundled polyclipping. I use same ebuild but can't reproduce too. Need more info.

**Comment by azhizhinov** [comment]
Hello.
Sorry for delay.
I'm not experienced enough to debug this issue, sorry.
So I have 2 laptops with the same hardware and this issue is reproducible always.
Both releases with the same issue.
Can this issue be hardware dependent somehow?
LibrePCB is built from sources.
https://mirror.yandex.ru/gentoo-distfiles/distfiles/librepcb-0.1.5-source.zip
https://mirror.yandex.ru/gentoo-distfiles/distfiles/librepcb-0.1.6-source.zip
Resulting packages are.
[librepcb-0.1.5.tbz2.zip](https://github.com/LibrePCB/LibrePCB/files/8885864/librepcb-0.1.5.tbz2.zip)
[librepcb-0.1.6.tbz2.zip](https://github.com/LibrePCB/LibrePCB/files/8885873/librepcb-0.1.6.tbz2.zip)
![20220612 LIBREPCB ORG 00](https://user-images.githubusercontent.com/13044908/173234746-d85758a8-d9a8-4ea3-a7ff-b6442577e6b7.jpg)
![20220612 LIBREPCB ORG 01](https://user-images.githubusercontent.com/13044908/173234751-05540c79-7ff8-46aa-bf1d-f49f17a01aa8.jpg)
![20220612 LIBREPCB ORG 02](https://user-images.githubusercontent.com/13044908/173234756-0b802913-c15a-497d-b330-457167061735.jpg)

**Comment by azhizhinov** [comment]
Hello.
Big thanks for Viktor K (ktrace).
The reason was found.
Both releases were built with these flags:
```
Compiler flags:
C -march=haswell -O2 -pipe
C++ -march=haswell -O2 -pipe
```
Today, as Victor K advised, I've re-compiled 0.1.6 release with these flags:
```
C -O2 -march=x86-64 -pipe
C++ -O2 -march=x86-64 -pipe
```
And now plates are filled as expected. Yay.
So, your code is very fragile.
![20220614 LIBREPCB ORG 00](https://user-images.githubusercontent.com/13044908/173572651-20497faf-c153-4322-a33a-bb40a3e2198d.jpg)
![20220614 LIBREPCB ORG 01](https://user-images.githubusercontent.com/13044908/173572695-951e5294-2822-4578-b506-c2f6a439c838.jpg)
![20220614 LIBREPCB ORG 02](https://user-images.githubusercontent.com/13044908/173572712-7fe8af7e-2592-4f5f-a574-4483b09212c8.jpg)

**Comment by ubruhin** [comment]
Awesome, thanks a lot for providing these details! With `-march=haswell` I was able to reproduce the bug and after quite some debugging I saw that different floating point optimizations lead to a slightly negative number instead of zero, which was passed to `qSqrt()` and then messed up the plane calculations :sob:

Fixed it in #991 :rocket:

**Comment by azhizhinov** [comment]
Hello.That's awesome, thanks a lot.
